### PR TITLE
[mc-159] create storage poc

### PR DESCRIPTION
**Pull request by eliasscaceres**
## What did you implement:

Closes #XXXXX

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [ ] Write tests and confirm existing functionality is not broken.
       **Validate via `npm test`**
- [ ] Write documentation
- [ ] Ensure there are no lint errors.
       **Validate via `npm run lint-updated`**
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [ ] Ensure introduced changes match Prettier formatting.
       **Validate via `npm run prettier-check-updated`**
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

**_Is this ready for review?:_** NO
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by FabioOrlinski** [comment]
Looks good!

I test it with and without the name param.

With name:
image

Without name:

image

**Review by taten — APPROVED** [review]
Nice one! 🚀

File: samples/gcp-storage/src/storageHandler.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const { streamToString } = require("./streamToString");
+
+const app = new App(new GcpModule());
+
+module.exports.handler = app.use(
+  [HTTPBindingMiddleware(), StorageMiddleware()],
+  async (context) => {
+    const { name } = context.req.query;
+    const opts = {
+      container: "storage_sample123",
```
Comment (FabioOrlinski) [inline samples/gcp-storage/src/storageHandler.js]: The container name is set here and in the yalm. Could you add it to a config file?

File: gcp/src/gcpModule.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
-import { CloudModule, CloudContext, CloudRequest, CloudResponse, ComponentType, CloudService } from "@multicloud/sls-core";
+import {
+  CloudModule,
+  CloudContext,
+  CloudStorage,
+  CloudRequest,
+  CloudResponse,
+  ComponentType,
+  CloudService,
+} from "core";
```
Comment (FabioOrlinski) [inline gcp/src/gcpModule.ts]: In the other files, you were using @multicloud/sls-core

File: samples/gcp-storage/src/storageHandler.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const { GcpModule } = require("gcp");
+const { streamToString } = require("./streamToString");
+const { bucketName } = require("../config.google.json");
+
+const app = new App(new GcpModule());
+
+module.exports.handler = app.use(
+  [HTTPBindingMiddleware(), StorageMiddleware()],
+  async (context) => {
+    const { name } = context.req.query;
```
Comment (FabioOrlinski) [inline samples/gcp-storage/src/storageHandler.js]: The query type is StringParams:

image

You could change it to `const { name } = context.req.query.toJson()`?

image

**Review by FabioOrlinski — CHANGES_REQUESTED** [review]

File: samples/gcp-storage/src/storageHandler.js (hunk 3 above)
Comment (FabioOrlinski) [reply samples/gcp-storage/src/storageHandler.js]: Done in a8a0c1e7acf6e7e3f64dfe26c570ef0224198c8d

File: gcp/src/gcpModule.ts (hunk 2 above)
Comment (FabioOrlinski) [reply gcp/src/gcpModule.ts]: Done in 64ab69892f5c3ab2cf2a4b060f3d9dc162266d82

File: samples/gcp-storage/src/storageHandler.js (hunk 1 above)
Comment (FabioOrlinski) [reply samples/gcp-storage/src/storageHandler.js]: Done in 64ab69892f5c3ab2cf2a4b060f3d9dc162266d82

**Review by FabioOrlinski — APPROVED** [review]

**Pull request merged**